cli/listener/src/main.rs

```diff
@@ -430,43 +432,83 @@ type DateBeginEnd = Range<chrono::DateTime<chrono::Utc>>;
 fn get_date_params<'a, I: IntoIterator<Item = (Cow<'a, str>, Cow<'a, str>)>>(
     query_params: I,
 ) -> Result<DateBeginEnd, ResponseError> {
-    let mut begin_opt: Option<chrono::DateTime<chrono::Utc>> = None;
-    let mut end_opt: Option<chrono::DateTime<chrono::Utc>> = None;
+    let mut begin_opt: Vec<chrono::DateTime<chrono::Utc>> = vec![];
+    let mut end_opt: Vec<chrono::DateTime<chrono::Utc>> = vec![];
+
     for (k, v) in query_params {
         match k.as_ref() {
             "begin" => {
                 let t = parse_utc_iso8601(&v).map_err(|_| {
                     ResponseError::BadRequest(
                         "Invalid ISO 8601 timestamp in 'begin' query parameter. \
-                    Expected e.g. '2022-02-15T23:59:59+00:00'.",
+                    Expected e.g. '2022-02-15T23:59:59Z'.",
                     )
                 })?;
-                begin_opt = Some(t);
+
+                begin_opt.push(t);
             }
             "end" => {
                 let t = parse_utc_iso8601(&v).map_err(|_| {
                     ResponseError::BadRequest(
                         "Invalid ISO 8601 timestamp in 'end' query parameter. \
-                    Expected e.g. '2022-02-15T23:59:59+00:00'.",
+                    Expected e.g. '2022-02-15T23:59:59Z'.",
```

Comment:
The previous example is still valid, right?

Reply:
I'm not really sure, it was giving me the error when I used the date format from the example

Reply:
It should accept both.

Reply:
Yeah but it's throwing error `Invalid ISO 8601 timestamp in 'begin' query parameter. Expected e.g. '2022-02-15T23:59:59+00:00'.` for me when I try this on browser 
https://solana.testnet.fi/api/apy/apy?begin=2022-02-04T11:40:02.683960+00:00&end=2022-02-07T14:22:08.826526+00:00

Reply:
That’s because you are not url-escaping the +, :, and . in the url, try this:

https://solana.testnet.fi/api/apy/apy?begin=2022-02-04T11%3A40%3A02.683960%2B00%3A00&end=2022-02-07T14%3A22%3A08.826526%2B00%3A00

Reply:
Maybe we can add that caveat to the error message.

Reply:
Ohhh, I thought browser would do that for me. 

> Maybe we can add that caveat to the error message.

 what should be the error message?

Reply:
> what should be the error message?

I would change it from

> Invalid ISO 8601 timestamp in 'begin' query parameter. Expected e.g. '2022-02-15T23:59:59+00:00'.

to

> Invalid ISO 8601 timestamp in 'begin' query parameter. Expected e.g. '2022-02-15T23:59:59+00:00'. Note that query parameters must be url-encoded.